composio/local_tools/local_workspace/workspace/actions/create_workspace.py

```diff
@@ -1,57 +1,211 @@
+class CreateWorkspaceResponse(BaseWorkspaceResponse):
     workspace_id: str = Field(..., description="workspace-id for the created workspace")
 
 
-class CreateWorkspaceAction(Action):
+class CreateWorkspaceAction(BaseWorkspaceAction):
     """
-    creates a workspace, and returns workspace-id
+    Creates a workspace, and returns workspace-id
     """
 
     _display_name = "Create workspace"
     _request_schema = CreateWorkspaceRequest
     _response_schema = CreateWorkspaceResponse
-    _tags = ["workspace"]
-    _tool_name = "localworkspace"
-    workspace_factory: WorkspaceManagerFactory = None
-    history_processor: HistoryProcessor = None
 
     def execute(
         self, request_data: CreateWorkspaceRequest, authorisation_data: dict
     ) -> CreateWorkspaceResponse:
+        if self.workspace_factory is None:
+            raise ValueError("Workspace factory is not set")
+
+        if request_data.image_name == "":
+            request_data.image_name = "sweagent/swe-agent:latest"
+
         args: LocalDockerArgumentsModel = LocalDockerArgumentsModel(
             image_name=request_data.image_name
         )
         workspace_id = self.workspace_factory.get_workspace_manager(args)
+        self.workspace_id = workspace_id
+        workspace_meta = get_workspace_meta_from_manager(
+            self.workspace_factory, self.workspace_id
+        )
+        if not workspace_meta:
+            raise ValueError(
+                f"workspace not found, invalid workspace-id: {self.workspace_id}"
+            )
+        self.container_name = workspace_meta[KEY_CONTAINER_NAME]
+        self.image_name = workspace_meta[KEY_IMAGE_NAME]
+        self.container_process = get_container_process(
+            workspace_meta[KEY_WORKSPACE_MANAGER]
+        )
+        self.parent_pids = workspace_meta[KEY_PARENT_PIDS]
+        self.container_obj = get_container_by_container_name(
+            self.container_name, self.image_name
+        )
+        self.return_code = None
+        self.logger = logger
+        self.config_file_path = script_dir / Path("../../config/default.yaml")
+        self.config = AgentConfig.load_yaml(self.config_file_path)
+        if not self.container_obj:
+            raise ValueError(
+                f"container-name {self.container_name} is not a valid docker-container"
+            )
+        self.set_env_variables()
+        env_vars = "\n".join(self.config.env_variables)
+        print(f"environment is set with variables: {env_vars}")
         return CreateWorkspaceResponse(workspace_id=workspace_id)
 
-    def set_workspace_and_history(
-        self,
-        workspace_factory: WorkspaceManagerFactory,
-        history_processor: HistoryProcessor,
-    ):
-        self.workspace_factory = workspace_factory
-        self.history_processor = history_processor
+    def set_env_variables(self):
+        commands_to_execute = (
+            [self.config.state_command.code]
+            + ["pip install flake8"]
```

Comment:
is flake8 installed in workspace for any specific purpose? The create workspace can be faster if we remove it